# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1456107482 -39600
#      Mon Feb 22 13:18:02 2016 +1100
# Node ID 4f0f9f029d127b70dd7b896031b61187d3ff3851
# Parent  7ba48ff6ce83f549fdf2293ae9d085ee5fe9f830
Draw updates to GUI at end of GUI callback function (bug #47216)

* graphics.cc (do_execute_callback): Fdrawnow if Vdrawnow_requested.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -9334,16 +9334,22 @@ gh_manager::do_execute_callback (const g
         {
           std::string nm = cb.class_name ();
           error ("trying to execute non-executable object (class = %s)",
                  nm.c_str ());
         }
 
       if (fcn)
         feval (fcn, args);
+
+      if (Vdrawnow_requested)
+        {
+          Fdrawnow ();
+          Vdrawnow_requested = false;
+        }
     }
 }
 
 void
 gh_manager::do_post_event (const graphics_event& e)
 {
   event_queue.push_back (e);
 
